core/include/jiminy/core/stepper/AbstractStepper.h

```diff
@@ -0,0 +1,82 @@
+            /// \param[in] robots Robots whose dynamics the stepper will work on.
+            AbstractStepper(F1 & f,
+                            std::vector<Robot const *> robots);
+
+            /// \brief Attempt to integrate the system from time t to time t + dt.
+            /// \details Each stepper is responsible for implementing its own error checking routine to determine
+            ///          if an integration step was successful or not. On sucess, the value of the state is
+            ///          updated in place. Regardless of success or failure, the parameter dt can be updated
+            ///          by variable step schemes to indicate the next advised dt.
+            ///
+            /// \param[in]      t     Integration start time.
+            /// \param[in, out] q     System starting position.
+            /// \param[in, out] v     System starting velocity.
+            /// \param[in, out] a     System starting acceleration.
+            /// \param[in, out] dt    Input: desired integration duration. Output: recommanded step size for
+            ///                       variable-step schemes. Constant-step schemes leave this value unmodified.
+            /// \return True if integration was successful, false otherwise. In that case, (q, v, a) are not updated.
+            virtual bool try_step(float64_t const& t,
+                                  std::vector<vectorN_t> & q,
+                                  std::vector<vectorN_t> & v,
+                                  std::vector<vectorN_t> & a,
+                                  float64_t & dt) = 0;
+
+        private:
+            /// \brief Increment the given position by the given velocity, respecting the Lie group structure.
+            /// \details addToVelocity(q, v * dt) returns exp(v * dt) * q, where exp is the exponential map
+            ///          of the Lie group.
+            ///          More simply, this can be viewed for vector spaces as computing q + v * dt, i.e.
+            ///          updating q by the desired velocity.
+            /// \param[in] positionIn Starting position
+            /// \param[in] velocityIn Desired velocity increment
+            /// \return Incremented position
+            std::vector<vectorN_t> addToPosition(std::vector<vectorN_t> const & positionIn, std::vector<vectorN_t> const& velocityIn);
+
+            /// \brief Increment the given velocity by the given acceleration for all system.
+            /// \details addToVelocity(v, a * dt) simply computes v + a * dt for all systems.
+            /// \param[in] velocityIn Starting velocity
+            /// \param[in] accelerationIn Desired acceleration increment
+            /// \return Incremented velocity
+            void addToVelocity(std::vector<vectorN_t> const & velocityIn, std::vector<vectorN_t> const& accelerationIn);
+
+
+            std::conditional_t<std::is_function<F1>::value,
```

Comment:
`private`

Reply:
f_ needs to be accessed by sub-classes as well - not robots_.

Reply:
indeed